### ISSUE TITLE
Feedback from users requested

### ISSUE THREAD

**Issue by cmallwitz** [cmallwitz/Financials-Extension#10]
Please leave feedback here about what platforms you use the extension on, what you use it for, what symbols/exchanges are important to you and what you think is missing.

**Comment by omoknen** [comment]
LibreOffice Windows 10  (used mortee instructions in another post).
Using it for keeping track of some standard mutual funds which works great.  Thanks for creating the extension.

**Comment by ikyquocked** [comment]
Windows 7. LibreOffice 7.0.4.2 is autocorrecting any digit with the comma in the beginning ex. ",21" to "0,21" and then fx shows error:501. The letter ";" works fine. Its better to correct the readme manual and write to you ";" instead ",".

**Comment by ikyquocked** [comment]
Also Im notice the bug,
=GETREALTIME("MCX:MOEX";21;"GOOGLE") returns the value 15726, this is result without comma, real value is 157,26 rub.

**Comment by ikyquocked** [comment]
Also noticed with =GETREALTIME("TBIOA.ME";21;"yahoo") it returns error "Yahoo.getRealtime(TBIOA.ME, 21) - process: 'currency'"

**Comment by ikyquocked** [comment]
Hi, its me again =)
Noticed some strange auto-update behavior on specific sheet.
When I edit any cage, even an empty one, the waiting-cursor appear for few minutes and all stocks data starting updating (I see it in trace.log). Even I click to the other sheet it starts update, quite unconfortable and very slow. Of course, I am not pressing ctrl-shift-F9 but it works like some auto-update. Its more usable if its updating only by command...
Attaching a video illustration and the table here.
[Strange sheet.zip](https://github.com/cmallwitz/Financials-Extension/files/5928404/Strange.sheet.zip)



https://user-images.githubusercontent.com/63292793/106952967-e278c480-6742-11eb-94b8-9d82ec24419f.mp4

**Comment by cat-herder** [comment]
Thank you for this nice extension.  I use it with LibreOfficeDev 7.1.2 x64 on Windows 10.  One thing that would be great is that I see that Yahoo has Option data, so given a stock symbol, date, Call or Put, and strike price it would be possible to get a quote from Yahoo for an option.  They structure the option symbol in this way: (here is one for Ford) F210219C00012000 (Stock Symbol, 2 digit year, 2 digit month, 2 digit day, C for call or P for put, then the strike price with varying zeroes before and after depending on the number of digits in the strike price.  I do not know if you trade options but it would be nice to get some options data in an automated way such as Bid, Ask, Change%, Volume, Open Interest, and Implied Volatility.  Thanks a lot.

**Comment by cmallwitz** [comment]
I think it should be possible to support options at least a little. Looking at e.g. https://uk.finance.yahoo.com/quote/F210219C00012000 it seems the data is just in different fields. But I can't see interest and volatility details. Additionally, you as the user would need to pass the option symbol used by Yahoo to the extension function as it would be impossible to calculate them without a lot of additional data (starting e.g. trading calendars etc)

But interesting stuff nonetheless. Do you have another source of this information - as in a different web site? I have always looked at Bloomberg and Reuters web sites for comparison when working out what is valid/invalid but I can't seem to see any publicly available option data. Maybe it is time to add an optional paid-for data provider?

Christian

**Comment by cmallwitz** [comment]
Funnily I see option data provided by one of the German banks I deal with occationaly: https://www.comdirect.de/inf/search/optionsscheine.html?IC_BYPASS=1&SEARCH_VALUE=FORD+CALL+02&SEARCH_REDIRECT=true&REDIRECT_TYPE=FULLTEXT

If you click on the links you see quite a few data point for these options (for various strikes). German as the language used by the web site wouldn't be deterrent to add stuff like this to the extension but the prices from Germany (in Euro) may not be of much use to you. Maybe you have a bank or brokerage web site in the US publishing data like this to entice people trading/brokering with them?

**Comment by cat-herder** [comment]
Yes, barchart.com is a site that provides option chains also https://www.barchart.com/stocks/quotes/F/options
They have an API but it is not free sadly.  A paid API is probably the best way to go for options data since it is so complex due to the varying option chains and dates.  I usually just grab the info I need from my Brokerage and add it to my spreadsheet manually when it comes to options.

**Comment by cmallwitz** [comment]
I added some support for options - have a look at the new version and the examples.ods - it has an example for an option too.

**Comment by cat-herder** [comment]
> I added some support for options - have a look at the new version and the examples.ods - it has an example for an option too.

I will check it out here shortly. Appreciate your work on the extension.

**Comment by cat-herder** [comment]
> I added some support for options - have a look at the new version and the examples.ods - it has an example for an option too.

I have implemented the new Financials_Extension into my options spreadsheet and it is working very well.  I've built a formula to construct a properly formatted option symbol for Yahoo from the existing data in my spreadsheet and everything works nicely with the GETREALTIME function.  A huge timesaver, thank you!

**Comment by cmallwitz** [comment]
Little update for options (because I hadn't come accros this on Yahoo before): one can access an overview of available options e.g. for IBM like this: https://finance.yahoo.com/quote/IBM/options?p=IBM&straddle=true

**Comment by ruimnetob** [comment]
Hi using this on:
Libreoffice About:
Version: 6.4.6.2
Build ID: 1:6.4.6-0ubuntu0.20.04.1
CPU threads: 12; OS: Linux 5.4; UI render: default; VCL: gtk3; 

Python3:
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0] on linux

Ubuntu 20.04.2 LTS

Followed the Readme but can't run the unittests as expected. Getting following error:

======================================================================
ERROR: test_ft (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_ft
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/toor/Financials-Extension/src/test_ft.py", line 17, in <module>
    import financials
  File "/home/toor/Financials-Extension/src/financials.py", line 22, in <module>
    import unohelper
  File "/usr/lib/python3/dist-packages/unohelper.py", line 19, in <module>
    import uno
  File "/home/toor/.local/lib/python3.8/site-packages/uno/__init__.py", line 4, in <module>
    from base import Element, Css, Payload, UnoBaseFeature, UnoBaseField
ModuleNotFoundError: No module named 'base'


======================================================================
ERROR: test_google (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_google
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/toor/Financials-Extension/src/test_google.py", line 17, in <module>
    import financials
  File "/home/toor/Financials-Extension/src/financials.py", line 22, in <module>
    import unohelper
  File "/usr/lib/python3/dist-packages/unohelper.py", line 19, in <module>
    import uno
  File "/home/toor/.local/lib/python3.8/site-packages/uno/__init__.py", line 4, in <module>
    from base import Element, Css, Payload, UnoBaseFeature, UnoBaseField
ModuleNotFoundError: No module named 'base'


======================================================================
ERROR: test_yahoo (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_yahoo
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
    module = self._get_module_from_name(name)
  File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/home/toor/Financials-Extension/src/test_yahoo.py", line 19, in <module>
    import financials
  File "/home/toor/Financials-Extension/src/financials.py", line 22, in <module>
    import unohelper
  File "/usr/lib/python3/dist-packages/unohelper.py", line 19, in <module>
    import uno
  File "/home/toor/.local/lib/python3.8/site-packages/uno/__init__.py", line 4, in <module>
    from base import Element, Css, Payload, UnoBaseFeature, UnoBaseField
ModuleNotFoundError: No module named 'base'


Should I open an issue, or maybe someone has a tip?

**Comment by ruimnetob** [comment]
Hi using this extension for personal usage. Just trying to get Beta from Yahoo and found that doesnt work properly on some symbols i.e. "XLF".

My system: 

Libreoffice About:
Version: 6.4.6.2
Build ID: 1:6.4.6-0ubuntu0.20.04.1
CPU threads: 12; OS: Linux 5.4; UI render: default; VCL: gtk3; 

Python3:
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0] on linux

Ubuntu 20.04.2 LTS

Followed the Readme but can't run the unittests as expected. Should I open an issue, or maybe someone has a tip? Thank you. Error follows below.


 ```
 ======================================================================
  ERROR: test_ft (unittest.loader._FailedTest)
  ----------------------------------------------------------------------
  ImportError: Failed to import test module: test_ft
  Traceback (most recent call last):
    File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
      module = self._get_module_from_name(name)
    File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
      __import__(name)
    File "/home/toor/Financials-Extension/src/test_ft.py", line 17, in <module>
      import financials
    File "/home/toor/Financials-Extension/src/financials.py", line 22, in <module>
      import unohelper
    File "/usr/lib/python3/dist-packages/unohelper.py", line 19, in <module>
      import uno
    File "/home/toor/.local/lib/python3.8/site-packages/uno/__init__.py", line 4, in <module>
      from base import Element, Css, Payload, UnoBaseFeature, UnoBaseField
  ModuleNotFoundError: No module named 'base'
  
  
  ======================================================================
  ERROR: test_google (unittest.loader._FailedTest)
  ----------------------------------------------------------------------
  ImportError: Failed to import test module: test_google
  Traceback (most recent call last):
    File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
      module = self._get_module_from_name(name)
    File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
      __import__(name)
    File "/home/toor/Financials-Extension/src/test_google.py", line 17, in <module>
      import financials
    File "/home/toor/Financials-Extension/src/financials.py", line 22, in <module>
      import unohelper
    File "/usr/lib/python3/dist-packages/unohelper.py", line 19, in <module>
      import uno
    File "/home/toor/.local/lib/python3.8/site-packages/uno/__init__.py", line 4, in <module>
      from base import Element, Css, Payload, UnoBaseFeature, UnoBaseField
  ModuleNotFoundError: No module named 'base'
  
  
  ======================================================================
  ERROR: test_yahoo (unittest.loader._FailedTest)
  ----------------------------------------------------------------------
  ImportError: Failed to import test module: test_yahoo
  Traceback (most recent call last):
    File "/usr/lib/python3.8/unittest/loader.py", line 436, in _find_test_path
      module = self._get_module_from_name(name)
    File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
      __import__(name)
    File "/home/toor/Financials-Extension/src/test_yahoo.py", line 19, in <module>
      import financials
    File "/home/toor/Financials-Extension/src/financials.py", line 22, in <module>
      import unohelper
    File "/usr/lib/python3/dist-packages/unohelper.py", line 19, in <module>
      import uno
    File "/home/toor/.local/lib/python3.8/site-packages/uno/__init__.py", line 4, in <module>
      from base import Element, Css, Payload, UnoBaseFeature, UnoBaseField
  ModuleNotFoundError: No module named 'base'
  
```

**Comment by cmallwitz** [comment]
Maybe a installation problem of the LibeOffice SDK ? You seem to find the unohelper (in system path) but the uno package itself is coming from local directory (/home/toor/.local/) ?

Just a note: Google is broken - they change the whole website - all the Google tests will fail therefore...

**Comment by ruimnetob** [comment]
> Maybe a installation problem of the LibeOffice SDK ? You seem to find the unohelper (in system path) but the uno package itself is coming from local directory (/home/toor/.local/) ?
> 
> Just a note: Google is broken - they change the whole website - all the Google tests will fail therefore...

Hi @cmallwitz , thank you for the tip. I can confirm that the problem was I had done a 
`pip3 install uno`

And the library was conflicting with the correct one. It then started throwing issues in librecalc itself and crashing with std:bad_alloc ...

Fixed.

Works perfectly fine in my setup.

**Comment by logos88** [comment]
Thank you for this great extension. I use it with LibreOffice 7.1.5.2 on Arch Linux, and mainly to get info to calculate intrinsic value and to compare stocks.

To calculate intrinsic value in different ways (FCF, Dividends, EPS) it would be useful if we could get this values:
- FCF TTM (ex: https://finance.yahoo.com/quote/MMM/cash-flow?p=MMM)
- Diluted EPS (ttm): Is more accurate than normal EPS (ex: https://finance.yahoo.com/quote/MMM/key-statistics?p=MMM)
- Trailing Annual Dividend Rate (ex: https://finance.yahoo.com/quote/MMM/key-statistics?p=MMM)

It would be interesting to have some growth rates like 5-Year Annual Average Net Income or 5-Year Annual Average Dividends from https://www.msn.com/en-us/money/stockdetails/analysis/fi-a1xroc

An finally, a 5 years P/E average would be also awesome to compare with present valuations. I have found MSN has P/E Ratio 5-Year High and P/E Ratio 5-Year Low so it could be easy to calculate the average: https://www.msn.com/en-us/money/stockdetails/analysis/fi-a1xroc

Morningstar also has 5Y Avg: http://financials.morningstar.com/valuation/price-ratio.html?t=MMM&region=usa&culture=en-US

**Comment by kastaldi** [comment]
I'm using this extension on Windows 10 x64 and LibreOffice 7.2.0.4 x64, just simple realtime stock prices to keep an eye on my portfolio. No issues until now but I had to use ";" instead of "," in Usage github section to make it work.
Thanks for the great job.

**Comment by cmallwitz** [comment]
> To calculate intrinsic value in different ways (FCF, Dividends, EPS) it would be useful if we could get this values:
> ...

I could certainly take a look to make more data points available but in general I try to have similar coverage between Yahoo and FT to keep them interchangable. Adding more providers has been suggested as well but the problem here is whether they have similar coverage fo rasset classes (FX, equity, funds, etc)

**Comment by jbourvic** [comment]
Just happened upon your post that you left to a search for Does LO support stock searches. Have a Win10 x64. Wasn't too sure about the proper loading of the extension (couldn't find 'Tools, Extensions' under Calc, but when I downloaded and ran your extension, it appeared to load and that was it. So hesitantly, I tried the command on a new page and BOOYAH! It worked! Even substituting a cell  for the name. You are the MAN! I so appreciate it. I have been searching for so long for something like this. Thanks.

**Comment by CBFR** [comment]
Hello,  I am a retired accountant and for years have struggled finding a complete stock quote tool. I have always used windows throughout my professional life. I have spent ages withe Excel which gets more and more complicated in a quest for sophistication and have given up. I then settled for Google Sheets which is not comprehensive and not totally reliable. I have now the time to explore the open source world. Your SIMPLE and FAST and apparently COMPREHENSIVE solution is just what I have been looking for. Now I can rebuild my  portfolio in Calc and I think it will be a great time-saver for me.  I will update when I have finished my Portfolio. Thank you for this extension.  A big plus for me is that I can even use this in privacy .

**Comment by CBFR** [comment]
I have now reworked my portfolios and I had to write a few macros (which was challenging) since FT does not give historic info.  It all works fine using the FT source. I cannot seem to get the Yahoo source to work. Here is the trace log extract which shows "r" which I guess must be the share price as "none". I also attach the ft version which correctly picks up the price:
2021-12-14 19:09:20.451205 getRealtime *args=('GSK:LSE', 21.0, 'ft') r='1598.0' 0.083 ms
2021-12-14 19:09:39.169863 getRealtime *args=('GSK:L', 21.0, 'yahoo') r='None' 673.045 ms
I would be grateful for some help.

**Comment by CBFR** [comment]
Thank you for the prompt reply but correcting the : to . still appears not to work. I must be missing something.

2021-12-14 22:12:44.762468 getRealtime *args=('GSK.L', 21.0, 'yahoo') r='None' 197.849 ms

**Comment by cmallwitz** [comment]
Hmm - not sure what is happening here... Using `=GETREALTIME("GSK.L", 21, "YAHOO")` creates the same log line on my side just with a proper value (instead on None)

**Comment by CBFR** [comment]
Please let me know if I can furnish any more info to help.....

**Comment by cmallwitz** [comment]
Can you try to put the formula in an otherwise empty spread sheet or open the file [examples.ods](https://github.com/cmallwitz/Financials-Extension/blob/master/examples.ods?raw=true) ? The later has Yahoo and FT examples - just to make sure it is working there...

**Comment by CBFR** [comment]
Thank you for your prompt reply
I loaded examples.ods and still had same error - All works with FT but with Yahoo I get "#N/D" message
Similarly when I opened a blank file and tried.

I notice in the trace log below that the middle term "21" is in fact picked up as "21.0". This also happens with requests for FT  but FT requests are all working fine nonetheless.

I am running on Windows 10 in France.
I have tried changing the  country and regional settings in Windows from France to Great Britain and changing the decimal point from "," (as in France) to ".".  However there is no change - I still have FT working and Yahoo showing #N/D.

2021-12-16 21:18:20.196447 getRealtime *args=('IBM', 21.0, 'Yahoo') r='None' 576.701 ms
2021-12-13 13:29:32.737615 getRealtime *args=('EURGBP', 21.0, 'ft') r='0.8504' 392.884 ms

**Comment by kastaldi** [comment]
Thanks for the extension. Prices weren't loading anymore but I upgraded from 3.0.3 to 3.0.6 and the problems were gone.

**Comment by CBFR** [comment]
I have just upgraded from 3.0.5 to 3.0.6 and Yahoo and FT are both working well.

**Comment by spacecatz03** [comment]
Hi,
I've been using this extension for a while in a spreadsheet and it has been working great, thank you so much! This morning it was working fine as usual but now I'm getting an error, e.g. =GETREALTIME("IBM",21,"YAHOO") results in - Yahoo.getRealtime(IBM, 21) - urlopen: argument must be an int, or have a fileno() method. Usually, I would just need to try later or quit LibreOffice and re-open sheet. Now it doesn't seem to be pulling in the data. Using Win11, LibreOffice 7.3.3.2 and extension version 3.0.6.

**Comment by cmallwitz** [comment]
Confirmed - Yahoo not working for me neither. Let me take a look - hoping it is just a temporary glitch and not some web site re-design requiring more work...

**Comment by spacecatz03** [comment]
Cool, thanks for the quick response.

**Comment by spacecatz03** [comment]
I'm getting this error when trying to install    (com.sun.star.uno.RuntimeException) { { Message = "<class 'FileNotFoundError'>: [Errno 2] No such file or directory: 'C:\\\\Users\\\\robma\\\\AppData\\\\Roaming\\\\LibreOffice\\\\4\\\\user\\\\uno_packages\\\\cache\\\\uno_packages\\\\lu603238whp.tmp_\\\\Financials-Extension%20(1).oxt\\\\financials.py', traceback follows\X000a  File \"C:\\Program Files\\LibreOffice\\program\\pythonloader.py\", line 146, in writeRegistryInfo\X000a    mod = self.getModuleFromUrl( locationUrl )\X000a  File \"C:\\Program Files\\LibreOffice\\program\\pythonloader.py\", line 93, in getModuleFromUrl\X000a    with open( filename, encoding='utf_8' ) as fileHandle:\X000a\X000a", Context = (com.sun.star.uno.XInterface) @0 } }

**Comment by spacecatz03** [comment]
Sorry, my bad. Once I completely uninstalled the previous version it worked! Thanks again, you da man!

**Comment by cmallwitz** [comment]
No worries - the way LibreOffice handles extensions is a bit brittle - e.g. if your rename the downloaded extension file ie. from Financials-Extension.oxt to Financials-Extension-3.0.7.oxt the installation will fail :-(

**Comment by TheSerapher** [comment]
Hey @cmallwitz,

Just noticed that quotes are not refreshing and are showing N/A for my sheet. Using MacOS and not sure how to get you any logs but maybe something changed on Yahoo's end again?

**Comment by cmallwitz** [comment]
Thanks for the notification - yes - it is broken for me too! 
I would certainly hope they just have messed with there cookies again instead of breaking the web site for us - I will update the extension accordingly. Maybe I need to extend the code a bit to fetch/update cookies as required but I don't want to add additional network round trips for all Yahoo request all the time...
Cheers!

**Comment by TheSerapher** [comment]
Can extensions be integrated to the Libreoffice settings? It may help to have a sane default that works at the time of writing but some ability to overwrite this setting by simply putting that new cookie string into the settings page. I have mine working again by adjusting the above cookies.

**Comment by cmallwitz** [comment]
Pushed new version 3.0.8

**Comment by TheSerapher** [comment]
Removed the old (modified) version 3.0.7 and installed the new 3.0.8: Works 👍

**Comment by SoftwareExplorerMe** [comment]
Worked great for getting Yahoo last price for a couple dozen stocks into my calc spreadsheet. But when I tried to add a few new stock tickers by copy/pasting the formula of a working cell into the formula bar and editing just the stock sticker it fails and says #NAME? in the cell but the older cells still provide updated info. EXAMPLE:
=ROUND(AutoAddIn.Financials.getRealtime("CHMI",21,"Yahoo"),4)    ==> COPY CELL and PASTE becomes
=ROUND(AutoAddIn.Financials.getRealtime("CHMI",21,"Yahoo"),4)    and works in new cell too BUT
=ROUND(AutoAddIn.Financials.getRealtime("CHMI",21,"Yahoo"),4)    ==> COPY FORMULA and PASTE in new cell formula=>
=ROUND(autoaddin.financials.getrealtime("CHMI",21,"Yahoo"),4)      loses Capitalization and results in #NAME? in cell

Win 10 PRO 1H22 and 2H22 LibreOffice 7.4.2.3 64bit (recently updated)

I'm not sure but I thought the formulas said =GETREALTIME... until after the LibreOffice update that changed it to 
=AutoAddIn.Financials.getRealtime...

Any clues as to how to fix?

**Comment by cmallwitz** [comment]
I would usually assume lower/upper case function names doesn't make a difference. I haven't tested the very latest LibreOffice - maybe they have changed how functions are addressed in formulae but would be surprised they did and broke anything. Have you tried removing the 'AutoAddIn.Financials' bit and see if it still works?

**Comment by TheSerapher** [comment]
Looks like YT Financial changed something again, not getting any data. Replacing the cookies didn't seem to work this time (maybe wrong files or something) but it looks like a few cookies, including EuConsent, have new data.

**Comment by TheSerapher** [comment]
Since I felt that YT was unreliable and I liked having up-to-date quotes for my cryptos at least, I went ahead and did this: https://github.com/cmallwitz/Financials-Extension/pull/67

**Comment by cmallwitz** [comment]
Arrggghhh - I was planning to spend some time over Christmas to put in place a more long-term/dynamic solution for the Yahoo cookie issue but they seem to have decided to create a (pre-)Christmas present for us... I will take look.

**Comment by cmallwitz** [comment]
Published new version 3.0.9 in the meantime

**Comment by spacecatz03** [comment]
Awesome, thanks for the quick fix much better than N/A# . Now just need an extension to keep the market from going down, lol.

**Comment by SoftwareExplorerMe** [comment]
Downloaded new 3.0.9 and installed (after removing 3.0.7) and Voila, my original sheet worked again with all Ticker symbols.
However the COPY/PASTE FORMULA Then changing Ticker Symbol in formula still doesn't work. But THANKFULLY, the  COPY/PASTE CELL DOES work for the existing Tickers. Just can't add in any new ones?!?!

**Comment by gmishkin** [comment]
> Version: 7.0.4.2
> Build ID: 00(Build:2)
> CPU threads: 4; OS: Linux 5.10; UI render: default; VCL: kf5
> Locale: en-US (en_US.UTF-8); UI: en-US
> Debian package version: 1:7.0.4-4+deb11u4
> Calc: threaded

Found on Stack Overflow

Using it for US mutual funds and ETFs

Any way we can support the project?

**Comment by spacecatz03** [comment]
May just be me but I think the Yahoo cookie issue is back, getting the dreaded #N/A again today.

**Comment by cmallwitz** [comment]
Yahoo deployed a major change to their website - can you try the most recent release 3.2.0 ?

**Comment by spacecatz03** [comment]
Oh, didn't realize the recent update. Yep, that works, thanks!

**Comment by TheSerapher** [comment]
Hey @cmallwitz,

I updated to 3.2.1 just now and most things work again but seeing this error for a few quotes:

*trace.log*
```
2023-01-05 11:18:06.335574 getRealtime *args=('0P000083LP.F', 21.0, 'YAHOO') r=''<' not supported between instances of 'float' and 'NoneType'' 0.090 ms
```

These are the entries in my table:

```
=GETREALTIME("0P000083LP.F";21;"YAHOO")
=GETREALTIME("UI3G.F";21;"YAHOO")
=GETREALTIME("U1IC.F";21;"YAHOO")
=GETREALTIME("U1IH.F";21;"YAHOO")
```

No cache files are written for these calls. Maybe something on Yahoo side changed with the format of the HTML that the scripts can't work with?

**Comment by cmallwitz** [comment]
Can you check the formulas? If I enter the formulas using ; they get converted to , (trace.log is independent of this)

Maybe Yahoo delivers you a different version of HTML than to me (your formulas work for me).

Could you attach (HOME directory) .financials-extension/yahoo-0P000083LP.F.html here ?

**Comment by cmallwitz** [comment]
Hmm - having just said that now I'm getting the error too - not that I know why :-(

**Comment by TheSerapher** [comment]
> Can you check the formulas? If I enter the formulas using ; they get converted to , (trace.log is independent of this)
> 

I think this is just a LibreOffice setting how to read formulas, shouldn't make any difference for the error message resulting.

> Maybe Yahoo delivers you a different version of HTML than to me (your formulas work for me).
> 
> Could you attach (HOME directory) .financials-extension/yahoo-0P000083LP.F.html here ?

Wish I could, but the calls fail before the HTML is cached, probably because the evaluation of the current value isn't working? Rather thought that because it was looking for a float but got nothing.

If you have an easy way for me to dump the HTML even though the error happens let me know. Then I could attach it.

**Comment by TheSerapher** [comment]
As a sidenote, seeing that Yahoo has extracted some information from HTML and added it to Javascript instead I raise the question if their goal is to make parsing really hard or next to impossible. It may be that they are actively trying to stop this kind of requests which would mean we'll have to find other means of fetching this data.

After my research there are only APIs available for paid services which is really annoying considering this is public information. So probably not a great option either.

This is speculation of course but maybe one that comes true.

**Comment by TheSerapher** [comment]
Switched my YT calls to the proper symbols on Financial Times and they work just fine. Also it's much faster, so I might stick to that.

Obviously for others the issue may still exist but I have found a workaround for it :)

**Comment by cmallwitz** [comment]
I will keep looking into this - it should save the HTML before parsing anything - so something isn't quite right.

Re: paid services - you are absolutely right - not an option for most people. Additionally even if a paid service has a "free" tier, they have the annoying habit of e.g. only allowing a fixed number of lookups in any given minute (let's say 5). This means you can't lookup your favourite 10 ETFs at once, even if you only do it once a day.

**Comment by alfonsi4** [comment]
I just installed it on Libre Office using Windows 10.... it works for what I need.

Thank You!

**Comment by Honminhui** [comment]
This is really great cmallwitz! One feature that might be nice is the ability to select the prior trading day in the case where the date selected is not a trading day in the historic data.

**Comment by OXOOOOE9XD** [comment]
Version: 7.4.3.2 (x64) / LibreOffice Community
Build ID: 1048a8393ae2eeec98dff31b5c133c5f1d08b890
CPU threads: 4; OS: Windows 10.0 Build 19044; UI render: Skia/Raster; VCL: win
Locale: en-IN (en_IN); UI: en-US
Calc: threaded

I installed the extension today to test. And endup with several errors while enabling it.

Following ZIP contain all the snapshot of error I capture.
[Finanace Errors.zip](https://github.com/cmallwitz/Financials-Extension/files/10420630/Finanace.Errors.zip)

I couldn't able to find log file in LibraOffice sorry for that.

**Comment by roadskater** [comment]
Thanks for a great extension! Perhaps this cell formula will be useful in the readme to help some whenever YAHOO results fail. It uses the data from YAHOO unless it is #N/A, in which case it uses the data from FT.

=IFNA(GETREALTIME("QQQ",21,"YAHOO"),GETREALTIME("QQQ",21,"FT"))

https://wiki.documentfoundation.org/Documentation/Calc_Functions/IFNA

**Comment by cmallwitz** [comment]
You certainly can check if you get the data you expect with IFNA but I would be careful with using a ticker symbol from Yahoo on FT. In general the symbols are slightly different and you may not get the data you are expecting either. Having said that at least for US equity FT seems to accept the ticker symbol and redirects to the main symbol for FT (this happens for QQQ in your example). Note: that requires an additional network roundtrip and makes downloading symbol data slower...

Take a look at the URLs below and experiment a little with the URL paramaters:

[https://finance.yahoo.com/quote/QQQ?p=QQQ](https://finance.yahoo.com/quote/QQQ?p=QQQ)
[https://markets.ft.com/data/etfs/tearsheet/summary?s=QQQ:NMQ:USD](https://markets.ft.com/data/etfs/tearsheet/summary?s=QQQ:NMQ:USD)

**Comment by SoftwareExplorerMe** [comment]
Strange thing that whenever I install a new extension version, the new tickers that I previously added that didn't work suddenly start working. But I still cannot add new tickers without getting #NAME? as the result. I even tried plain old "IBM" and it won't work. I tried uninstalling the extension then reinstalling , but that didn't fix it. I have 2 oddball tickers I tried to get from FT instead of Yahoo "OP:NSQ" & "ISO:NSQ" but that wouldn't work either. So keep up the good work??
US WIN 10 PRO LibreOffice 7.4.3.2 Extension 3.3.0 64bit

**Comment by roadskater** [comment]
Thank you for the response. I've updated my post, hopefully correctly, to reflect your comment.
    On Saturday, January 21, 2023 at 08:24:13 AM EST, cmallwitz ***@***.***> wrote:  
 
 


You certainly can check if you get the data you expect with IFNA but I would be careful with using a ticker symbol from Yahoo on FT. In general the symbols are slightly different and you may not get the data you are expecting either. Having said that at least for US equity FT seems to accept the ticker symbol and redirects to the main symbol for FT (this happens for QQQ in your example). Note: that requires an additional network roundtrip and makes downloading symbol data slower...

Take a look at the URLs below and experiment a little with the URL paramaters:

https://finance.yahoo.com/quote/QQQ?p=QQQ
https://markets.ft.com/data/etfs/tearsheet/summary?s=QQQ:NMQ:USD

**Comment by Futur3H4wk** [comment]
Thanks this is great! I was looking for information on how I might go about scraping info from CoinMarketCap or CopinGecko, into a spread sheet I'm using to track my crypto holdings. Found you had already done the heavy lifting. The only recommendation I would make is in your ReadMe file, you might make a mention that Yahoo also tracks crypto. I've never used Yahoo before and was disappointed that while CoinBase will display the prices for all my coins, I could not import the prices on any they don't currently have available on their platform (which was about half of them).

**Comment by cmallwitz** [comment]
> "OP:NSQ" & "ISO:NSQ"

I would expect "ISO:NSQ" to work but for OP do you mean "OP:NAQ" ?

**Comment by SoftwareExplorerMe** [comment]
> > "OP:NSQ" & "ISO:NSQ"
> 
> I would expect "ISO:NSQ" to work but for OP do you mean "OP:NAQ" ?

Yes, was a typo in my response. I did try it both ways though. Neither worked for either stock.
Still cannot add a new stock on sheet even IBM as a test.

**Comment by jdoss2020** [comment]
Thank you so much for developing this extension.  It works great for simply retrieving stock and mutual fund/ETF quotes from Yahoo.  It enabled me to ditch my Google sheets investment spreadsheet and implement the same functionality on my home linux system using libreoffice.  Bye, bye Google.  So far I've experienced no issues.  I really appreciate your efforts in developing and maintaining this solution.

**Comment by oscby** [comment]
Great extension, I use it tonnes. 
Adding a function to grab the next earnings date would be useful, yahoo finance provides it (some time in advance).

**Comment by TheSerapher** [comment]
Hey @cmallwitz,

Looks like Yahoo changed something again since I am unable to receive quotes from them. Haven't looked into it but if it's on their end you should be able to reproduce it.

**Comment by cmallwitz** [comment]
Yes - looks like it - I will take a look.

**Comment by jbourvic** [comment]
I find that they usually shut down access during weekends. No set time frame but later on everything starts working fine again.Sent via the Samsung Galaxy Note20 Ultra 5G, an AT&T 5G smartphone
-------- Original message --------From: Sebastian Grewe ***@***.***> Date: 7/3/23  03:46  (GMT-06:00) To: cmallwitz/Financials-Extension ***@***.***> Cc: jbourvic ***@***.***>, Comment ***@***.***> Subject: Re: [cmallwitz/Financials-Extension] General feedback from users
  requested (create new issue for specific questions) (#10) 
Hey @cmallwitz,
Looks like Yahoo changed something again since I am unable to receive quotes from them. Haven't looked into it but if it's on their end you should be able to reproduce it.

**Comment by cmallwitz** [comment]
I bumped up the cookies and it seems to be working again. Please try the latest version.

**Comment by TheSerapher** [comment]
Hey @cmallwitz,

Installed the released version but still seeing no results for my last two remaining YAHOO calls. All others work just fine.
When I open Yahoo I am able to find quotes for these two.

```
2023-07-06 08:34:58.025302 getRealtime *args=('IWDA.L', 21.0, 'YAHOO') r='None' 561.250 ms
[...]
2023-07-06 08:35:45.973728 getRealtime *args=('NU-EUR', 21.0, 'YAHOO') r='None' 217.603 ms
```

**Comment by estianic** [comment]
Here also no results from Yahoo calls, even after updating to version 3.3.1 with Libreoffice 7.3.7.2 

Spreadsheet shows #N/A as result. Logfile shows r='None' for all calls. Downloaded .html files can not be opened with browser, gets stuck at the point where you have to accept of refuse cookies (both choices have no effect). Downloaded .html files from a week ago can be opened.

**Comment by cmallwitz** [comment]
Can you try deleting (user home directory) .financials-extension / yahoo.cookiejar and then restart LibreCalc?

**Comment by estianic** [comment]
It works after deleting yahoo.cookiejar

Thanks!

**Comment by cmallwitz** [comment]
I checked both and I got values.

**Comment by DrakeBliss** [comment]
Great Extension! I am using it to track Dividend Stock. I am also wanting to track some ETF Funds. Yahoo does not have the dividend data for them. Theses for example do not work for getting annual dividend and ex-dividend dates.

https://finance.yahoo.com/quote/SVOL
https://finance.yahoo.com/quote/TSLY

I do see the data for annual dividend and ex-dividend date on this website

https://www.nasdaq.com/market-activity/etf/svol
https://www.nasdaq.com/market-activity/etf/tsly

Would it be a hard to add NASDAQ as one of the lookup providers?

Thank you for making this extension!

**Comment by Ulyzzeus** [comment]
> This is really great cmallwitz! One feature that might be nice is the ability to select the prior trading day in the case where the date selected is not a trading day in the historic data.

you can do this manually using `=IF(WEEKDAY(H82)=7;GETHISTORIC("EUR=X";"90";H82-1;"YAHOO");IF(WEEKDAY(H82)=1;GETHISTORIC("EUR=X";"90";H82-2;"YAHOO");GETHISTORIC("EUR=X";"90";H82;"YAHOO")))` where H82 is a cell with the date, but a shorter way would be nice

**Comment by Futur3H4wk** [comment]
I've set up a spread sheet using =GETREALTIME("BTC-USD",21,"YAHOO") and =GETREALTIME(B12,21,"YAHOO") with B12 having a stock symbol. It has worked great for several months. The last few days I've been getting an error 
Yahoo.getRealtimeSummary(PRWCX, 21) - urlopen: argument must be an int, or have a fileno() method.
I'm not finding any answers on the internet. Was there an update LibreOffice that has caused this? is there a workaround?

**Comment by Futur3H4wk** [comment]
I did a kernel update and rebooted my server and VBox. now everthing is working again.

**Comment by Durhammer3111** [comment]
I have NO COMPLAINT about this extension, it is FABULOUS. However, it would be interesting to see if there is a way to get the Yahoo! "Fair Value" information on a given ticker? As an example, currently TSLA shows it is "Near Fair Value" with a return of +11%. Would love to be able to pull that type of thing in. Thanks!

**Comment by Mago-actor** [comment]
I loved it, it is very powerful, congratulations. Just missing a user manual to facilitate it with all its possibilities. For example, I have not yet discovered how to see Euro Dollar values of other dates, not just the current one. But it is an excellent extension. Thank you.

**Comment by svg1234** [comment]
I joined github today just to give a heads up to anyone else who might encounter the same problem that I did in trying to get the extension to work. Using Arch and the AppImage of LO v7.6.1.2. I could not compile extension from scratch because I don't have LO nor the dev tools installed. I just downloaded the latest releasse from here. I managed to track down the problem. Despite the error message indicating the problem is unohelper, it's really that it's looking for **libcrypt.so.1** nad currently Arch is up to libcrypt.so.2.0.0. 

While it's not a great workaround, I merely created a softlink from what was expected to what is there, and presto, the extension works. But I will have to create a new link whenever a system update results in an updated libcrypt. Not sure exactly where the problem originates from? Where is the call to the old version of libcrypt coming from? Could this issue be addressed in future releases of the extension? Or is it the LO AppImage? Thank you.

**Comment by Durhammer3111** [comment]
Things have been going well until late this afternoon, when trying to get Calc to run calculations (Ctrl+Shift+F9). The cursor spun briefly, and the sheet was flooded with junk. Repeated attempts did the same. Wondered if Yahoo! was having issues, so waited and opened up an old sheet and got a bit less of the same thing (way simpler sheet setup). The main thing I see is this:

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0 Transitional//EN">

<html>
<head>
	
	
	<title></title>
	<meta name="generator" content="LibreOffice 7.0.4.2 (Linux)"/>
	<style type="text/css">
		body,div,table,thead,tbody,tfoot,tr,th,td,p { font-family:"Arial"; font-size:x-small }
		a.comment-indicator:hover + comment { background:#ffd; position:absolute; display:block; border:1px solid black; padding:0.5em;  } 
		a.comment-indicator { background:red; display:inline-block; border:1px solid black; width:0.5em; height:0.5em;  } 
		comment { display:none;  } 
	</style>
	
</head>

<body>

Yahoo.getRealtimeSummary(SMCI, 67) - urlopen: argument must be an int, or have a fileno() method. |   |   |   |   |   |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | --



</body>

</html>

What's the generally recommended debugging approach? Is it generally this?

> Can you try deleting (user home directory) .financials-extension / yahoo.cookiejar and then restart LibreCalc?

**Comment by cmallwitz** [comment]
There is no real debugging approach other than what you have quoted and then check the files.

Can you try the latest extension version to see if it makes a difference?

22 Jan 2024 23:16:22 Durhammer3111 ***@***.***>:

**Comment by H0lderness** [comment]
Hello @ cmallwitz

I have just started using the Financial Extension with LibreOffice Calc and I find the spreadsheet very slow to load. BUT the extension works and downloads prices OK it just takes a long time'

I am using Windows 10 latest version, 16 gb Ram, Processor AMD Ryzen 5 2600 Six-Core Processor, 3400 Mhz, 6 Core(s), so an inadequately powered PC should not be an issue.

I am not overly technical but his is what I did:-

Created two separate spread sheets one for FT and the other for Yahoo

I used two of the functions as shown in your webpage ie

=GETREALTIME("EURUSD","LAST_PRICE","FT")

=GETREALTIME("IBM",21,"YAHOO")

I inputted on each sheet Name, Ticker symbol and Price as headers, then inputted various securities, Shares, Investment Trusts, Mutual Funds and one Bond and one Index 32 in all. I then put the formula in each of the cells under price, just changing the Ticker symbol to the ones I wanted
Everything went fine first time round on each sheet all Securities Index and Bond prices were fetched correctly. It is just a) so long to load the spreadsheet b) so slow to fetch the prices. Since then it is repeatable but SLOW.

Do you have any suggestion as to why this is happening and what I can do about it?

Thank you

Terry

**Comment by Futur3H4wk** [comment]
Same situation here. I did recalc 1 cell and it populated (others did not) and the next time I did a hard recalc it went back to na. I expect the problem is on the yahoo side of things. had a similar situation a few moths ago and it corrected it self after a few days.

**Comment by Futur3H4wk** [comment]
also H0lderness it always take quite a while to update a bunch of cells. Mine takes 3 to 5 minuets with about 35 cells to update.

**Comment by Durhammer3111** [comment]
@Futur3H4wk, how are your calls arranged, per ticker? I now have two versions of my spreadsheet, one where there is one ticker per row with numerous different attribute calls, and another sheet where there is one ticker per column, with the same stock attribute calls. Both have over 60 tickers. The row-oriented sheet takes 10-15 minutes to update, but the column-oriented sheet takes 2 to 3 minutes. Big difference. I also learned to minimize the attributes so they can all be satisfied by the stock's summary file alone. Looking at the trace.log file showed me these things.

⁣Get BlueMail for Android ​

**Comment by Futur3H4wk** [comment]
In columns across several sheets. There are other calculations that are updated at the same time, some of them using the results of the calls. It’s not a big deal waiting a few minutes, I usually just play a game while I wait, unless I have something more important to do then I just look at it when I come back. I track crypto, stocks, IRAs and precious metals, all through Yahoo Finance (no subscription required), and I only check it a few times a day at most. This little piece of code is Fabu.

**Comment by H0lderness** [comment]
Hi @  Futur3H4wk


Arranged as below, horizontally. 35 securities. Based on what you say I will play about with my format.

Thanks for your input

abrdn Asia Pacific Equity Fund I Acc | Fund | GB00B0XWNG99:GBP | 319.1900
abrdn Emerging Markets Equity Fund I Acc | Fund | GB0033227561:GBP | 786.9100
Abrdn UK Smaller Companies Growth Trust plc | Investment Trust | AUSC:LSE | 438.0000
Alliance Trust PLC | Investment Trust | ATST.L | 1146.0000
Artemis SmartGARP European Equity Fund Class I Acc | Fund | GB00B2PLJD73:GBP | 495.0500
AstraZeneca PLC | Share | AZN.L | 10586.0000
AXA Framlington UK Select Opportunities ZI Acc | Fund | 0P0000XMRR.L | 174.3000
BlackRock World Mining Trust plc | Investment Trust | BRWM.L | 538.0000
Family UK Equity Life Fund | Bond | 0P0000JRBC | 1.6000
Invesco Perpetual Asian Y Acc | Fund | 0P00012AOT.L | 496.4200
Invesco China Equity Fund Y Acc | Fund | 0P00012ARQ.L | 282.7800
Invesco European Equity Y Acc | Fund | 0P00012AP9.L | 396.2400
Invesco Global Equity Fund Y Acc | Fund | 0P00012APK.L | 495.6000
Invesco Global Smaller Companies Fund Y Acc | Fund | 0P00012APR.L | 457.8500
Invesco Rspnb Jpn Equ Val Discovery Fund (UK) Y Acc | Fund | 0P00012ARV.L | 335.4500
Invesco UK Eq High Inc UK Y Acc | Fund | 0P00012APU.L | 237.3900
Invesco UK Equity Inc UK Y Acc | Fund | 0P00012ART.L | 234.8300
JPM Emerging Markets B Net Acc | Fund | 0P0000K7VW.L | 344.8000
JPMorgan Emerging Markets Inv Trust | Investment Trust | JMG:LSE | 102.6000
Jupiter European Special Sits I Acc | Fund | 0P0000U20H.L | 531.0100
Jupiter Income Trust I Inc | Fund | 0P0000U20O.L | 551.0600
Jupiter UK Smaller Companies Eq I Acc | Fund | 0P0000X1GJ.L | 356.9500

**Comment by Futur3H4wk** [comment]
@SoftwareExplorerMe There is an updated version of the extention. It fixed my problem after I also updated my version of LibreOffice.

**Comment by H0lderness** [comment]
Hi @ Futur3H4wk

The Financials Extension currently on GitHub  is v 3.4 (its a fortnight old)

When I reported my problems I was already using v 3.4. So no change

**Comment by Futur3H4wk** [comment]
@ H0lderness Other folks got it to work by uninstalling the old version, rebooting then installing the new version. That didn't work for me I also had to update my LibreOffice. I went from 7.3 (Distro Default) to 24.2 (FlatPak). Now it's working.

**Comment by cmallwitz** [comment]
> Hi @ Futur3H4wk
> 
> Arranged as below, horizontally. 35 securities. Based on what you say I will play about with my format.
> 
> Thanks for your input
> 
> abrdn Asia Pacific Equity Fund I Acc | Fund | GB00B0XWNG99:GBP | 319.1900 abrdn Emerging Markets Equity Fund I Acc | Fund | GB0033227561:GBP | 786.9100 Abrdn UK Smaller Companies Growth Trust plc | Investment Trust | AUSC:LSE | 438.0000 Alliance Trust PLC | Investment Trust | ATST.L | 1146.0000 Artemis SmartGARP European Equity Fund Class I Acc | Fund | GB00B2PLJD73:GBP | 495.0500 AstraZeneca PLC | Share | AZN.L | 10586.0000 AXA Framlington UK Select Opportunities ZI Acc | Fund | 0P0000XMRR.L | 174.3000 BlackRock World Mining Trust plc | Investment Trust | BRWM.L | 538.0000 Family UK Equity Life Fund | Bond | 0P0000JRBC | 1.6000 Invesco Perpetual Asian Y Acc | Fund | 0P00012AOT.L | 496.4200 Invesco China Equity Fund Y Acc | Fund | 0P00012ARQ.L | 282.7800 Invesco European Equity Y Acc | Fund | 0P00012AP9.L | 396.2400 Invesco Global Equity Fund Y Acc | Fund | 0P00012APK.L | 495.6000 Invesco Global Smaller Companies Fund Y Acc | Fund | 0P00012APR.L | 457.8500 Invesco Rspnb Jpn Equ Val Discovery Fund (UK) Y Acc | Fund | 0P00012ARV.L | 335.4500 Invesco UK Eq High Inc UK Y Acc | Fund | 0P00012APU.L | 237.3900 Invesco UK Equity Inc UK Y Acc | Fund | 0P00012ART.L | 234.8300 JPM Emerging Markets B Net Acc | Fund | 0P0000K7VW.L | 344.8000 JPMorgan Emerging Markets Inv Trust | Investment Trust | JMG:LSE | 102.6000 Jupiter European Special Sits I Acc | Fund | 0P0000U20H.L | 531.0100 Jupiter Income Trust I Inc | Fund | 0P0000U20O.L | 551.0600 Jupiter UK Smaller Companies Eq I Acc | Fund | 0P0000X1GJ.L | 356.9500

Looks like you are pulling mostly (?) UK listed securities - you might be better off using FT rather than Yahoo. I find (being located in UK too) FT is quicker probably because it is closer network-wise.

**Comment by H0lderness** [comment]
Hi Futur3H4wk

I have tried using FT.com for Fund prices with the new Financials Ext v 3.4 and it is just as slow as the Yahoo one. (with this extension)

By contrast if you use the Lemon Fool Simple Yahoo price scrape (for Calc) with a mixture of stocks funds and others, 35 in all, it is lightning fast, 2 or 3 seconds to update. So the issue is the extension/ link/formulae.

**Comment by SoftwareExplorerMe** [comment]
YAHOO broken again today. All tickers come up N/A (with new 3.4 Version) It was working in AM but died PM.

**Comment by Futur3H4wk** [comment]
Still working for me. v3.4 LibreOffice 24.2.0.3

**Comment by Durhammer3111** [comment]
Same here -- worked right up to close of market (4PM), whereupon it went to Hades in a hand basket. Maybe it was simply a Yahoo server outage -- I'll have to test again in a few.

⁣Get BlueMail for Android ​

**Comment by Honminhui** [comment]
I really appreciated our friend, but his real passion was freedom.
Libreoffice with this extension continues his tradition
https://github.com/cmallwitz/Financials-Extension

-------- Original Message --------

**Comment by Honminhui** [comment]
I really appreciated our friend, but his real passion was freedom.
Libreoffice with this extension continues his tradition
https://github.com/cmallwitz/Financials-Extension

-------- Original Message --------

**Comment by SoftwareExplorerMe** [comment]
Yahoo was working later in PM . Guess it was just Yahoo's servers temporary problem.

**Comment by spacecatz03** [comment]
Yes, it seems the data pull hasn't been the most reliable, was down a bit this morning for me, then back working again.

**Comment by SoftwareExplorerMe** [comment]
Yeah, stopped working for me last 2 days since.

**Comment by Futur3H4wk** [comment]
Got the NA thing tonight, 1st time since the update of both extention and LbreCalc. The disable-restart-enable-restart worked like a charm.

**Comment by Durhammer3111** [comment]
The extension (or Yahoo) has been failing on me for a day now. Tried @Futur3H4wk 's "disable-restart-enable-restart" and its more severe variant (remove/reinstall) to no end. There's no Yahoo cookie file to delete anymore, so that's no help. Some curious findings in the logs and directory:

First, all the "yahoo-XXXX.html" files are only between 57 and 63 bytes in size. Some messages from the two log files:

extension.log --

2024-03-05 11:11:17,373 financials_yahoo ERROR BaseException ticker=DECK datacode=27
Traceback (most recent call last):
  File "/home/xxxxx/.config/libreoffice/4/user/uno_packages/cache/uno_packages/lu1768699ekfy4z.tmp_/Financials-Extension.oxt/financials_yahoo.py", line 201, in getRealtimeSummary
    found = root.findall(f".//fin-streamer[@data-symbol='{ticker}']")
AttributeError: 'NoneType' object has no attribute 'findall'

trace.log at various times:

2024-03-05 12:23:37.576698 getRealtime *args=('DECK', 27.0, 'YAHOO') r='Yahoo.getRealtimeSummary(DECK, 27) - urlopen: argument must be an int, or have a fileno() method.' 14.406 ms

2024-03-05 12:23:37.718403 getRealtime *args=('DECK', 69.0, 'YAHOO') r=''<' not supported between instances of 'float' and 'NoneType'' 0.134 ms

2024-03-05 18:09:50.446099 getRealtime *args=('DECK', 27.0, 'YAHOO') r='Yahoo.getRealtimeSummary(DECK, 27) - process: 'NoneType' object has no attribute 'findall'' 256.773 ms

Of course, these all could (must?) simply be from the html files being almost empty.

Suggestions on how to get out of this?

**Comment by Futur3H4wk** [comment]
The issue of being unable to pull from Yahoo is being discussed in 2 threads. It would be good to consolidate in https://github.com/cmallwitz/Financials-Extension/issues/98

**Comment by H0lderness** [comment]
Hi@cmallwitz et al

FINANCIALS-EXTENSION

I hope this is the right part of the forum to ask this question?

I am using the LibreOffice Extension for Calc for this site and the following formula for getting data from FT.com:-

=GETREALTIME("IBM:NYQ",21,"FT")

It works well apart from two issues when I boot the spreadsheet up 1) it takes a long time. and 2) every time it boots it updates prices (presumably that's why it takes a long time). There are times when I want to open the file, but don't want prices updated.

Can anyone explain to me how to turn off or on the updating process?

Thank you for your help

Terry

**Comment by H0lderness** [comment]
Hi Durhammer3111

Thanks so much for the information. Really appreciated

Terry

**Comment by sergioantoy** [comment]
Donwloaded version 3.5.0 on 03/13/2024. Running with libreoffice 7.0.4.2, python 3.9.2, debian 11. It works well. Thank you.

Sergio

**Comment by DMel74** [comment]
FYI v3.5.0 running on Win11, LibreOffice 24.2.2

**Comment by spacecatz03** [comment]
I'm getting the dreaded #N/A today. Using latest version of extension. For example, =GETREALTIME("AAPL",21,"YAHOO") returns #N/A. Double-checked internet connection, restarted LIbreOffice, etc. no joy.

**Comment by spacecatz03** [comment]
FYI - rolled back to v3.5.0 and now it works again.

**Comment by 4044ever** [comment]
FYI v3.6.0 running on Win11, LibreOffice 7.6.7.2
32Gb RAM

Yahoo - not working
FT - working

Performance is very slow.

**Comment by SoftwareExplorerMe** [comment]
Upgraded to 3.6.1 after seeing Durhammer3111 post. Much faster now. But Yahoo displayed only a few values yesterday and now all are "Value!"

**Comment by H0lderness** [comment]
Hi

V3.6.1 working OK on pulling prices from Yahoo here in UK. Just tried it 0732 UK time 31/05/2024

**Comment by SoftwareExplorerMe** [comment]
It started working again so must have been a temporary glitch with Yahoo. When I say faster I mean it now takes just a few seconds rather than about 30+

**Comment by Durhammer3111** [comment]
I have been using FT for some time now, but figured I ought to go back and try my Yahoo! version. It's not working. I am on MX Linux 21.3, LO 7.0.4.2, and the Financials Extension 3.6.1 . Everything seems to be getting "crumb missing" messages.

**Comment by SoftwareExplorerMe** [comment]
Been working great now on Yahoo WIN 10 PRO LO 24.2.1.2 Fin Ext 3.6.1



=ROUND(AUTOADDIN.FINANCIALS.GETREALTIME("TICKER",21,"Yahoo"),4)




</body>

</html>

**Comment by H0lderness** [comment]
Hi Durhammer3111

Just updated 0742 UK time - 32 securities a mix of stocks and funds. No problems with v3.6.1 on Windows Home.

**Comment by H0lderness** [comment]
Hi

Just tried to update this morning Saturday 0813 hours UK time. Yahoo prices not working. It says against all prices "Crumb missing"

Anyone else having problems?

Thanks.

**Comment by H0lderness** [comment]
Hi

See my previous post.

I am using Financial Extension v3.6.1 May

**Comment by H0lderness** [comment]
Hi 

See my previous posts. It seems to have corrected itself and is now downloading OK. Why, I don't know.

**Comment by H0lderness** [comment]
Hi @ Smallwitz

I am getting problems with downloading from Yahoo using latest extension 3.6.1. Error message says crumb missing. Please see image. 

Any ideas?

Thanks
![ErrorMessage Yahoo Smallwitz](https://github.com/user-attachments/assets/57117246-9b74-4806-88a7-557a4b8d7464)

**Comment by H0lderness** [comment]
I should have added that I am using Windows 10 latest version.

Terry

**Comment by Durhammer3111** [comment]
After seeing these last reports from @H0lderness , I figured it was time to revisit my issues I reported back in June. To wit,

> I have been using FT for some time now, but figured I ought to go back and try my Yahoo! version. It's not working. I am on MX Linux 21.3, LO 7.0.4.2, and the Financials Extension 3.6.1 . Everything seems to be getting "crumb missing" messages.

I've been using the FT calls since before then, and up until today. So of course today when I tried the Yahoo! calls, it all is working for whatever reason(s). While I don't like the iffy flakiness of the Yahoo! calls, it's a lot less of a hassle than having to deal with setting up the correct FT exchange tickers and the difference in certain fields. So crazy that it all can turn on a dime, tho'....

Right now, the Yahoo calls are what I'm using and they are working fine, and are fast. Thanks, @cmallwitz !

**Comment by SoftwareExplorerMe** [comment]
Been having Yahoo issues recently also. Fix for me has been the old "uninstall extension & reinstall extension" method.

**Comment by H0lderness** [comment]
Hi @ Smallwitz et al

I have just tried removing and replacing the extension v3.6.1 and it doesn't make any difference. I am still getting the crumb missing messages.

**Comment by H0lderness** [comment]
Hi @ Smallwitz et al

Further to my recent post about not being able to download market prices from Yahoo and getting an error message - "Crumb missing"

After further work I can, on odd occasions, download prices, that said it is erratic. I am based in the UK I have tried everything from shutting firewall and antivirus down. In addition the suggestions made in this forum re disabling or removing and reinstalling the v3.6.1 extension. None of which works.

I would welcome any suggestions from participants and the extension author.

Thanks

**Comment by H0lderness** [comment]
Hi @ cmallwitz

Still the same problem all this week & today/

Terry

**Comment by estianic** [comment]
I get the "crumb missing" error too, but only once in a while. Like  one times out of ten. Closing Libreoffice and then restarting it usually solves the problem, removing and reinstalling the extension is not necessary. As far as I understand, it is caused by a communication problem with the Yahoo server.

I really can't say why you have this problem constantly. If the UK location has something to do with it, you could try from another country with a VPN.

**Comment by H0lderness** [comment]
Hi estianic

Thanks for your comments. It is interesting to know that I am not alone. I
wasn't aware of the Yahoo server comms. problem.

It has only been occurring for several months. Prior to that there was no
problem. In my case the problem is more frequent than yours, although there
are times when it works straight away.

Anyway, thanks for your input. It would be interesting to know if
the developer of the extension is looking at the problem.

Terry

**Comment by ElhemEnohpi** [comment]
I just started using this last week - very helpful, thanks! Except that today I was only getting the "crumb missing" errors, and no data. I tried various things, including recalculating with F9, and restarting LibreOffice, to no avail. But then later in the evening, restarting it again did work.

**Comment by askrates** [comment]
I tried installing today using the Financials-Extension.oxt, but the extensions installer hangs. I have to force-close Libreoffice. Am on Opensuse Tumbleweed, Libreoffice 24.8.0.3.

**Comment by helenkapol** [comment]
Everything is working well, but if it is possible add also ATR to the table. ATR - Average True Range . A very useful ratio

**Comment by Durhammer3111** [comment]
Aaaaarrrrgggghhhh. Back to using FT. Looks like Yahoo changed again, @cmallwitz -- looking at the files downloaded for YAHOO, all the .html files are from today (Sep. 14) but all the .json files are from yesterday (Sep. 13).

**Comment by H0lderness** [comment]
Hi @ Smallwitz

I have had a problem this morning with a fund not downloading from FT.com. First time this has ever happened with this fund. Yet the same fund is showing priced correctly on FT.com

Name of Fund [Legal & General European Index Trust I Class Distribution](https://markets.ft.com/data/funds/tearsheet/summary?s=GB00B0CNGQ43:GBP)

Ticker Symbol GB00B0CNGQ43:GBP

What I am using in in the Price Column to call the price:- =GETREALTIME("GB00B0XWNG99:GBP ",21,"FT")

I am getting #N/A

Any chance you could check it for me

Thanks

**Comment by cmallwitz** [comment]
=GETREALTIME("GB00B0XWNG99:GBP ",21,"FT") seems to be working for me right now - I get 356.06 
Maybe an intermittent weekend issue?

**Comment by H0lderness** [comment]
Hi @ cmallwitz

Thanks very much for your confirmation. 

I don't think it's anything to do with your extension because I have just found out that the same thing happens on a different piece of software, Simple Price Scrape from Lemon Fool.

I did by chance find a solution by removing the ":GBP" or ".GBX which leaves the basic ISIN ticker symbol. This works with your extension.

I cannot explain why this is the case, but, I note that a number of funds are moving towards pricing in pounds and the ":GBP" or ".GBX are FT.Com designations. 

Anyway I am back up and running. Thanks for your help and have a good Christmas and New Year.

H0lderness

**Comment by iluvmyt250** [comment]
thank you for this nice extension, i'm using it on debian 11,  libreoffice 7.0.4.2  , and it works fine so far and it was easy to get going.      Not doing anything fancy , in my spreadsheet I have some stocks, and I use the extension to grab real time stock prices so I can do a graph within calc showing how much of a position in dollars I have in each stock.

**Comment by Z10b06duck-com** [comment]
Yahoo has been down for 2 days for me too. I uninstalled 3.7 and installed 3.7.2 and tried again. The other providers still work but Yahoo still seems to just time out ( spreadsheet takes longer to load ) and GETREALTIME returns nothing.

**Comment by Futur3H4wk** [comment]
upgraded to 3.7.2 as expected didn't fix the problem. I'm gussing it's on yohoo's end.

**Comment by Zab69** [comment]
Works fine on Windows 10 and Windows 11 with the Yahoo fix. On W11, the cmd instructions for install had to be modified a bit to work.

**Comment by thirzel** [comment]
Works perfectly for FT. Only issue: Yahoo Finance queries (e.g., =GETREALTIME("AAPL", 102, "YAHOO")) fail with "failed to fetch crumb" due to Yahoo’s updated CSRF protection. Happy to test a fix if you implement it!
